tests/child.cpp

```diff
@@ -204,6 +206,88 @@ TEST(childproc, ptrace_child_term_before_execve)
   EXPECT_EQ(child->term_signal(), 15);
 }
 
+TEST(childproc, multi_exec_match)
+{
+  std::string tmpdir = "/tmp/bpftrace-test-child-XXXXXX";
+
+  // Create directory for test
+  if (::mkdtemp(&tmpdir[0]) == nullptr)
+  {
+    throw std::runtime_error("creating temporary path for tests failed");
+  }
+
+  const std_filesystem::path path(tmpdir);
+  const std_filesystem::path usr = path / "usr";
+  const std_filesystem::path usr_bin = usr / "bin";
+  const std_filesystem::path symlnk_bin = path / "bin";
+  const std_filesystem::path binary = usr_bin / "mysleep";
+
+  if (!std_filesystem::create_directory(usr))
+  {
+    throw std::runtime_error("creating subdirectory for tests failed");
+  }
+  if (!std_filesystem::create_directory(usr_bin))
+  {
+    throw std::runtime_error("creating subdirectory for tests failed");
+  }
+
+  // Create symbol link: bin -> usr/bin
+  char cwd[512];
+  if (::getcwd(cwd, sizeof(cwd)) == NULL)
+  {
+    throw std::runtime_error("Failed to getcwd().");
+  }
+  if (::chdir(path.c_str()))
+  {
+    throw std::runtime_error("Failed to chdir().");
+  }
```

Comment:
I don't think chdir is necessary, right? Can symlink two absolute paths

Reply:
If `EXPECT_EQ(::symlink(usr_bin.c_str(), symlnk_bin.c_str()), 0);`, will got this:
```
# sudo tree /tmp/bpftrace-test-child-*
/tmp/bpftrace-test-child-GIcs6T
├── bin -> /tmp/bpftrace-test-child-GIcs6T/usr/bin
└── usr
    └── bin
        └── mysleep
```

However, we should keep the 'rootfs' this way:

```
# sudo tree /tmp/bpftrace-test-child-*
/tmp/bpftrace-test-child-rMpfBC
├── bin -> usr/bin
└── usr
    └── bin
        └── mysleep
```

Like the `/` did (`bin -> usr/bin`):

```
$ ls -l /bin
lrwxrwxrwx. 1 root root 7 Aug 10  2021 /bin -> usr/bin
```
Thus, i just `chdir -> link -> chdir`. What do you think?